pkg/minikube/cluster/cluster.go

```diff
@@ -122,12 +132,27 @@ func DeleteHost(api libmachine.API) error {
 	if err != nil {
 		return err
 	}
+
+	if err = unregister(host); err != nil {
+		return err
+	}
+
 	m := util.MultiError{}
 	m.Collect(host.Driver.Remove())
 	m.Collect(api.Remove(constants.MachineName))
 	return m.ToError()
 }
 
+// Un-register host VM
+func unregister(host *host.Host) error {
+	if drivers.MachineInState(host.Driver, state.Running)() {
```

Comment:
I don't get it, so you want to rely on the code which checks whether un-registration is even required (based on the type of VM). AFAICT this code lives in _registration.UnregisterHostVM_ which inside the if statement checking whether the VM is running. 

> so if a vm is any other state than running it will display error about vm status and above command is not executable on vm.

If the VM is not running, the registrar check is not even occurring, right?

I guess that one could argue that the delete as whole will fail, due to _host.Driver.Remove()_ or _api.Remove(constants.MachineName)_ failing, but looking at the docs for _api.Remove_ it seems open how exactly the driver handles this.

Am I missing something here?